src/SFA.DAS.EmployerFinance/Services/DasLevyService.cs

```diff
@@ -1,99 +1,104 @@
-﻿using MediatR;
-using SFA.DAS.EmployerFinance.Models.Transaction;
-using SFA.DAS.EmployerFinance.Queries.AccountTransactions.GetAccountProviderPayments;
-using SFA.DAS.EmployerFinance.Queries.GetAccountTransactions;
-using SFA.DAS.EmployerFinance.Queries.GetPreviousTransactionsCount;
+﻿using SFA.DAS.EmployerFinance.Models.Transaction;
 using System;
 using System.Collections.Generic;
 using System.Linq;
 using System.Threading.Tasks;
 using SFA.DAS.EmployerFinance.Data;
-using SFA.DAS.EmployerFinance.Queries.AccountTransactions.GetAccountCoursePayments;
-using SFA.DAS.EmployerFinance.Queries.GetAccountLevyTransactions;
 
 namespace SFA.DAS.EmployerFinance.Services
 {
     public class DasLevyService : IDasLevyService
     {
-        private readonly IMediator _mediator;
         private readonly ITransactionRepository _transactionRepository;
+        private readonly IHmrcDateService _hmrcDateService;
 
-        public DasLevyService(IMediator mediator, ITransactionRepository transactionRepository)
+        public DasLevyService(ITransactionRepository transactionRepository, IHmrcDateService hmrcDateService)
         {
-            _mediator = mediator;
             _transactionRepository = transactionRepository;
+            _hmrcDateService = hmrcDateService;
         }
 
-        public async Task<ICollection<TransactionLine>> GetAccountTransactionsByDateRange(long accountId, DateTime fromDate, DateTime toDate)
+        public Task<decimal> GetAccountBalance(long accountId)
         {
-            var result = await _mediator.SendAsync(new GetAccountTransactionsRequest
-            {
-                AccountId = accountId,
-                FromDate = fromDate,
-                ToDate = toDate
-            });
+            return _transactionRepository.GetAccountBalance(accountId);
+        }
 
-            return result.TransactionLines;
+        public Task<TransactionLine[]> GetAccountTransactionsByDateRange(long accountId, DateTime fromDate, DateTime toDate)
+        {
+            return _transactionRepository.GetAccountTransactionsByDateRange(accountId, fromDate, toDate);
```

Comment:
This method gets called by EmployerAccountTransactionsOrchestrator which uses the PaymentDate L377.
Also is this paymentDate something that is could be used if someone directly called the repository as it may need to be moved if this is the case to make sure its always set even if you do not get transactionLines without the use of the DasService

Reply:
After discussing with Ian it was decided that this change should be investigated as a separate tech task to see if it is a bug that needs fixing or whether the code is meant to work in that way as the original code did not have the check.